mgmt/rest/v2/config.go

```diff
@@ -0,0 +1,158 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/cdata"
+	"github.com/intelsdi-x/snap/mgmt/rest/v2/response"
+	"github.com/julienschmidt/httprouter"
+)
+
+func (s *V2) getPluginConfigItem(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	var err error
+	styp := p.ByName("type")
+	if styp == "" {
+		cdn := s.configManager.GetPluginConfigDataNodeAll()
+		item := &response.PluginConfigItem{ConfigDataNode: cdn}
+		response.Write(200, item, w)
+		return
+	}
+
+	typ, err := getPluginType(styp)
+	if err != nil {
+		response.Write(400, response.FromError(err), w)
+		return
+	}
+
+	name := p.ByName("name")
```

Comment:
Do you mean we should skip the next block (verifying the version) if the name is empty?